### PR TITLE
Set custom text and color to bar boxes.

### PR DESCRIPTION
**Pull request by FakelsHub**
`sfall_func3("set_iface_tag_text", int tagId, string text, int color)`

Used colors:
0 - Green (default)
1 - Red
2 - White
3 - Yellow
4 - Dark Yellow (Peanut Butter)
5 - Blue
6 - Purple

### CODE REVIEW

**Comment by NovaRain** [comment]
Compilation errors aside, the code makes the game be terminated abruptly after loading. Have you compiled or tested the code before commit?

**Comment by FakelsHub** [comment]
Compilation error ??
In version 3.8, this works without failures. Send me the compiled dll.

**Comment by NovaRain** [comment]
>Compilation error ??

Line 163 in BarBoxes.cpp, and ResetBoxes() shouldn't be at the end of the code.
Also you should consider using SafeWriteBatch<T> template.

>In version 3.8, this works without failures. Send me the compiled dll.

OK, I found the cause: the length of text labels is limited to 15 characters. There's should be a way to trim the text or prevent the function being executed if the text is too long (printing errors or something).

**Comment by FakelsHub** [comment]
>  I found the cause: the length of text labels is limited to 15 characters.

I knew that there is a problem with long lines.
Added cutting long text.

File: sfall/Modules/BarBoxes.cpp [hunk 1]
```diff
@@ -34,6 +38,17 @@ struct sBox {
 static sBox boxes[10];
 static DWORD boxesEnabled[5];
 
+#define tSize    28
```
Comment (NovaRain) [inline sfall/Modules/BarBoxes.cpp]: why changing tSize to 28? Text length over 16 characters doesn't fit the box well.

Comment (FakelsHub) [reply sfall/Modules/BarBoxes.cpp]: characters have a different size, can fit up to 18 characters. made the buffer size 20 with a margin.

**Pull request merged**